File: IBA Test/Python ibacocktail/Cocktail.py
class Cocktail:

    def __init__(self, name, category, ingredients, method, garnish, history, note, simoxrate, link):
        self.name = name
        self.category = category
        self.ingredients = ingredients
        self.method = method
        self.garnish = garnish
        self.history = history
        self.note = note
        self.simoxrate = int(simoxrate)
        self.link = link


    #TODO

    def get_name(self):
        return self.name
    
    def get_category(self):
        return self.category

    def get_ingredients(self):
        return self.ingredients.copy()

    def get_method(self):
        return self.method

    def get_garnish(self):
        return self.garnish

    def get_history(self):
        return self.history

    def get_note(self):
        return self.note

    def get_simoxrate(self):
        return self.simoxrate

    def get_link(self):
        return self.link

    def containsIngredient(self, ingredient):
        for ingred in self.ingredients:
            if ingredient.lower() in ingred.get_i_name().lower():
                return True
        return False
    
    def compareTo(self, other):
        return other.get_simoxrate() - self.simoxrate

    def __str__(self) -> str:
        string = ""

        string += "/-/\n" + self.name + ", " + self.category + "\nINGREDIENTS\n"

        for ing in self.ingredients:
            string += ing.get_i_name() + " - " + ing.get_i_quantity() + "\n"

        string += "METHOD\n" + self.method + "\nGARNISH\n" + self.garnish + "\nHISTORY\n" + self.history + "\nNOTE\n" + self.note + "\nSIMOXRATE\n" + str(self.simoxrate) + "\nLINK\n" + self.link

        return string

    def toString(self):
        string = ""

        string += "/-/\n" + self.name + ", " + self.category + "\nINGREDIENTS\n"

        for ing in self.ingredients:
            string += ing.get_i_name() + " - " + ing.get_i_quantity() + "\n"

        string += "METHOD\n" + self.method + "\nGARNISH\n" + self.garnish + "\nHISTORY\n" + self.history + "\nNOTE\n" + self.note + "\nSIMOXRATE\n" + str(self.simoxrate) + "\nLINK\n" + self.link

        return string

    def hasHigherSimoxRate(self, other):
        if self.get_simoxrate() > other.get_simoxrate():
            return True
        else:
            return False

    def __lt__(self, other):
        return self.get_simoxrate() < other.get_simoxrate()

    

File: IBA Test/Java ibacocktail/Cocktail.java
import java.util.ArrayList;
import java.util.List;

public class Cocktail implements Comparable<Cocktail> {

    private final String name;
    private final Category category;
    private final List<Ingredient> ingredients;
    private final String method;
    private final String garnish;
    private final String history;
    private final String note;
    private final int simoxrate;
    private final String link;

    public Cocktail(
        final String name,  
        final Category category, 
        final List<Ingredient> ingredients,
        final String method,
        final String garnish,
        final String history,
        final String note,
        final int simoxrate,
        final String link
    ) {
        this.name = name;
        this.category = category;
        this.ingredients = ingredients;
        this.method = method;
        this.garnish = garnish;
        this.history = history;
        this.note = note;
        this.simoxrate = simoxrate;
        this.link = link;
    }

    @Override
    public String toString() {
        StringBuilder s = new StringBuilder();

        s.append("/-/\n" + name + ", " + category + "\nINGREDIENTS\n");

        for (int i = 0; i < ingredients.size(); i++) {
            s.append(ingredients.get(i).getname() + " - " + ingredients.get(i).getquantity() + "\n");
        }

        s.append("METHOD\n" + method + "\nGARNISH\n" + garnish + "\nHISTORY\n" + history + "\nNOTE\n" + note + "\nSIMOXRATE\n" + simoxrate + "\nLINK\n" + link);
    
        return s.toString();
    }

    public String toString2() {
        StringBuilder s = new StringBuilder();

        s.append("----------------------\n" + name + ", " + category + "\nINGREDIENTS\n");

        for (int i = 0; i < ingredients.size(); i++) {
            s.append(ingredients.get(i).getname() + " - " + ingredients.get(i).getquantity() + "\n");
        }

        s.append("METHOD\n" + method + "\nGARNISH\n" + garnish + "\nHISTORY\n" + history + "\nNOTE\n" + note + "\nSIMOXRATE\n" + simoxrate + "\nLINK\n" + link);
    
        return s.toString();
    }

    public String getname() {
        return name;
    }

    public Category getcategory() {
        return category;
    }

    public List<Ingredient> getingredients() {
        return new ArrayList<>(ingredients);
    }

    public String getmethod() {
        return method;
    }

    public String getgarnish() {
        return garnish;
    }

    public String gethistory() {
        return history;
    }

    public String getnote() {
        return note;
    }

    public int getsimoxrate() {
        return simoxrate;
    }

    public String getlink() {
        return link;
    }

    public boolean containsIngredient(String iname) {
        for (int i = 0; i < ingredients.size(); i++) {

            if (ingredients.get(i).getname().toLowerCase().contains(iname.toLowerCase())) {
                return true;
            }

            /*
            if (ingredients.get(i).getname().equals(iname)) {
                return true;
            }
            */
        }
        return false;
    }
    
    @Override
    public int compareTo(Cocktail o) {
        Cocktail other = (Cocktail) o;
        return other.getsimoxrate() - this.getsimoxrate();
    }
}